### PR TITLE
Auth and Messaging Interop for Swift

### PR DESCRIPTION
**Pull request by paulb777**
- Set up Auth and Messaging Interop for usage from Swift libraries via Swift Package Manager
- Port Functions Context tests that exercise the interop to Swift
- Fix exposed issues in library port
- CocoaPods podspec updates
- Completes porting the Functions unit tests to Swift

### CODE REVIEW

File: FirebaseFunctionsSwift/Tests/Unit/ContextProviderTests.swift [hunk 1]
```diff
@@ -12,181 +12,140 @@
...
+    let provider = FunctionsContextProvider(auth: auth, messaging: messagingFake, appCheck: nil)
+    let expectation = expectation(description: "Context should have auth keys.")
+    provider.getContext { context, error in
+      XCTAssertNotNil(context)
+      XCTAssertEqual(context.authToken, "token")
+      XCTAssertEqual(context.fcmToken, self.messagingFake.fcmToken)
+      XCTAssertNil(error)
+      expectation.fulfill()
+    }
+    waitForExpectations(timeout: 1.5)
```
Comment (ryanwilson) [inline FirebaseFunctionsSwift/Tests/Unit/ContextProviderTests.swift:49]: Does this timeout need to be so high compared to the others?

File: FirebaseAuthInterop.podspec [hunk 2]
```diff
@@ -0,0 +1,30 @@
+Pod::Spec.new do |s|
+  s.name             = 'FirebaseAuthInterop'
```
[the remark this answers is not in the document]
Comment (paulb777) [reply FirebaseAuthInterop.podspec:2]: Added #9325 to track

File: FirebaseFunctionsSwift/Tests/Integration/IntegrationTests.swift [hunk 3]
```diff
@@ -658,17 +660,21 @@ class IntegrationTests: XCTestCase {
...
+  }
+
   let token: String
 
   init(token: String) {
     self.token = token
   }
 
-  func getToken(forcingRefresh: Bool, callback: (String?, Error?) -> Void) {
+  func getToken(forcingRefresh: Bool, completion callback: (String?, Error?) -> Void) {
```
[the remark this answers is not in the document]
Comment (paulb777) [reply FirebaseFunctionsSwift/Tests/Integration/IntegrationTests.swift]: Done.

File: FirebaseFunctionsSwift/Tests/Unit/ContextProviderTests.swift (hunk 1 above)
Comment (paulb777) [reply FirebaseFunctionsSwift/Tests/Unit/ContextProviderTests.swift]: Direct port, but no reason to be different than the others.

**Review by ryanwilson — APPROVED** [review]

**Pull request merged**